### PR TITLE
Add tests for vagrant boxes

### PR DESCRIPTION
**Pull request by LKHN**
Added Vagrant box tests and documentation

Signed-off-by: Elkhan Mammadli <elkhan.mammadli@protonmail.com>

### CODE REVIEW

**Comment by srbala** [comment]
@LKHN is this for automated testing use in Linux environments?

File: tests/vagrant/test_vagrant.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        raise NotImplementedError(f'Unsupported Hypervisor {hypervisor}')
+
+
+def test_insecure_vagrant_ssh_pub_key(host):
+    """Only Vagrant insecure public key should be present"""
+    authorized_keys = host.file("/home/vagrant/.ssh/authorized_keys").content_string
+    if len(authorized_keys.splitlines()) == 1:
+        # SHA256 checksum of "$vagrant_insecure_pub_key vagrant insecure public key"
+        assert host.file(
+            "/home/vagrant/.ssh/authorized_keys").sha256sum == "9aa9292172c915821e29bcbf5ff42d4940f59d6a148153c76ad638f5f4c6cd8b"
```
Comment (srbala) [inline tests/vagrant/test_vagrant.py:68]: Just question, auto test id keys?

**Review by srbala — COMMENTED** [review]
LGTM, just questions for understanding

**Comment by LKHN** [comment]
> @LKHN is this for automated testing use in Linux environments?

You can run this for Libvirt boxes on Linux. VirtualBox, VMware vagrant boxes on Linux and Windows environments. Hyper-V on Windows. The test can be used as either manual or automated. For instance, We use this in our pipeline at https://github.com/AlmaLinux/alcib. See `tests/vagrant/Vagrant.md` for tests and instruction.

File: tests/vagrant/test_vagrant.py (hunk 1 above)
Comment (LKHN) [reply tests/vagrant/test_vagrant.py:68]: The vagrant box must only have the vagrant insecure SSH key's public part in its authorized_keys file.

**Comment by srbala** [comment]
I got that, order of check variants. If most frequest use is automated testing current order is ok. otherwise other or check should start 3rd party software before defaults to system native, just a point of view.

**Pull request merged**